src/vivarium/framework/dependency.py

```diff
@@ -0,0 +1,124 @@
+"""
+=================================
+Dependency Management in Vivarium
+=================================
+
+This module provides a tool to manage dependencies on resources within a
+``Vivarium`` simulation. These resources take the form of things that can be
+created and utilized by components, for example columns in the state table or
```

Comment:
I think the best thing to link here would be the top-level API reference (as opposed to a term in the glossary) but I'm not actually sure how yet